server/http/restapi/webhook.go

```diff
@@ -0,0 +1,68 @@
+package restapi
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+)
+
+func (a *restapi) handleWebhook(w http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+
+	appID := vars["app_id"]
+	if appID == "" {
+		httputils.WriteBadRequestError(w, errors.New("app_id not specified"))
+		return
+	}
+
+	whName := vars["name"]
+	if whName == "" {
+		httputils.WriteBadRequestError(w, errors.New("webhook name not specified"))
+		return
+	}
+
+	queryVars := req.URL.Query()
+	if len(queryVars["secret"]) != 1 {
+		httputils.WriteBadRequestError(w, errors.New("webhook secret was not provided"))
+		return
+	}
+
+	secret := queryVars["secret"][0]
+	if !a.validSecret(appID, secret) {
+		httputils.WriteBadRequestError(w, errors.New("webhook secret is not valid"))
+		return
+	}
+
+	var c interface{}
+	err := json.NewDecoder(req.Body).Decode(&c)
+	if err != nil {
+		httputils.WriteBadRequestError(w, err)
+		return
+	}
+
+	values := make(map[string]interface{})
+	values["payload"] = c
+
+	call := apps.CallRequest{
+		Values: values,
+		Context: &apps.Context{
+			AppID: apps.AppID(appID),
+		},
+		Type: apps.CallTypeSubmit,
+		Call: apps.Call{
+			Path: "/webhook-incoming",
```

Comment:
Should we allow custom webhook calls like we do with Bindings and Install?

Reply:
I can certainly add that, but @levb and I discussed getting this PR working with Zendesk and abstracting pieces later.  We should probably make sure that we agree with the `default` path name `webhook-incoming` for this PR

Reply:
What do y'all think?

Reply:
For the App, all webhooks are "incoming", whether they come from Mattermost or a 3rd party system.

2/5 we refer to 3rd party systems as "remote (3rd party)" and in the code/config just "remote".

Reply:
Customizing the Expand of the webhook call is going to be essential in the future, but ATM would create difficulties since Expand does not enforce bot permissions.

It may be easier if less performant to use the REST API in webhook handlers.

Webhook handlers should be invoked with a `Notify`, async.